pebbles/tests/test_flask_api.py

```diff
@@ -2151,6 +2151,106 @@ def test_user_fetch_instance_usage_stats(self):
             path='/api/v1/stats')
         self.assertStatus(response, 403)
 
+    def test_create_namespaced_data(self):
+        response = self.make_authenticated_admin_request(
+            method='POST',
+            path='/api/v1/namespaced_keyvalues',
+            data=json.dumps({'namespace': 'TestDriver', 'key': 'test_pool_vm_1', 'value': {"attr": "val"}})
+        )
+        self.assert_200(response)
+
+    def test_create_duplicate_namespaced_data(self):
+        response = self.make_authenticated_admin_request(
+            method='POST',
+            path='/api/v1/namespaced_keyvalues',
+            data=json.dumps({'namespace': 'TestDriver', 'key': 'test_pool_vm_2', 'value': {"attr": "val"}})
+        )
+        self.assert_200(response)
+
+        # The combination of namespace and key has to be unique
+        response = self.make_authenticated_admin_request(
+            method='POST',
+            path='/api/v1/namespaced_keyvalues',
+            data=json.dumps({'namespace': 'TestDriver', 'key': 'test_pool_vm_2', 'value': {"attr": "new_val"}})
+        )
+        self.assertStatus(response, 422)
+
+    def test_update_namespaced_data(self):
+        ns = "TestDriver"
+        key = "pool_vm_test_3"
```

Comment:
Scratch that. @Jyrsa : The key was there because I was changing the key in line 2192. We should be able to change the key , right ? Or Should we keep it protected ?